k8s/src/main/scala/io/buoyant/k8s/RouteManager.scala

```diff
@@ -0,0 +1,35 @@
+package io.buoyant.k8s
+
+import com.twitter.util._
+import io.buoyant.k8s.IstioPilotClient.RouteRuleConfig
+import java.util.concurrent.atomic.AtomicReference
+import istio.proxy.v1.config.RouteRule
+import io.buoyant.namer.RichActivity
+
+class RouteManager(api: IstioPilotClient, pollInterval: Duration) {
+  private[this] val states: Var[Activity.State[Map[String, RouteRule]]] = api.watchRouteRules(pollInterval).run.map {
```

Comment:
I think this can just be
```
private[this] val routeRules: Activity[Map[String, RouteRule]] = api.watchRouteRules(pollInterval).map(mkRouteMap)
```
although you'd lose the logging...